sabnzbd/interface.py

```diff
@@ -150,25 +167,43 @@ def internal_wrap(*args, **kwargs):
     return internal_wrap
 
 
-def check_access(access_type=4):
+def check_access(access_type: int = 4, warn_user: bool = False) -> bool:
     """Check if external address is allowed given access_type:
     1=nzb
     2=api
     3=full_api
     4=webui
     5=webui with login for external
     """
-    referrer = cherrypy.request.remote.ip
+    # Easy, it's allowed
+    if access_type <= cfg.inet_exposure():
+        return True
 
     # CherryPy will report ::ffff:192.168.0.10 on dual-stack situation
-    # It will always contain that ::ffff: prefix
-    range_ok = not cfg.local_ranges() or bool(
-        [1 for r in cfg.local_ranges() if (referrer.startswith(r) or referrer.replace("::ffff:", "").startswith(r))]
-    )
-    allowed = referrer in ("127.0.0.1", "::ffff:127.0.0.1", "::1") or range_ok or access_type <= cfg.inet_exposure()
-    if not allowed:
-        logging.debug("Refused connection from %s", referrer)
-    return allowed
+    # It will always contain that ::ffff: prefix, the ipaddress module can handle that
+    referrer = cherrypy.request.remote.ip
+
+    # Check for localhost
+    if referrer in ("127.0.0.1", "::ffff:127.0.0.1", "::1"):
+        return True
+
+    # No special ranged defined
+    is_allowed = False
+    if not cfg.local_ranges():
+        try:
+            is_allowed = ipaddress.ip_address(referrer).is_private
+        except ValueError:
+            # Something malformed, reject
+            pass
+    else:
+        is_allowed = bool(
+            [1 for r in cfg.local_ranges() if (referrer.startswith(r) or referrer.replace("::ffff:", "").startswith(r))]
```

Comment:
The replace part could be problematic with certain ipv6 addresses (unless cherrypy guarantees a fully written-out version of the address for ipv6, eliminating any chance of a '::' in the address). For example, someone configures `22ab:cd:ef:` as the local range, and the remote ip is something like `22::ffff:ab:cd:ef:12:34`. Limiting the replacing to addresses that aren't ipv6 would avoid this issue.

Separate form the above: comparing address ranges this way (starts with...) is yuck. Too bad it isn't in proper network notation so we could use ipaddress to check whether some address is in a given range.
```
>>> i = ipaddress.ip_address("4.3.2.1")
>>> n = ipaddress.ip_network("10.6.48.0/22")
>>> i in n
False
```

Reply:
This is the same as the code was before, so far it worked but I admit I have no clue why and how.. The ipaddress module does recognize this format with `::ffff:` at the start. So maybe we can use it? Although I'm not sure how, do you maybe know? :)

Reply:
If it starts with `::ffff:` and contains dots, converting using `ipaddress.ip_address(input).ipv4_mapped` seems to work.

For the local ranges, we could start accepting the standard notation (e.g. _192.168.0.0/24_) `ipaddress.ip_network` takes, and internally convert legacy settings (easy to detect, no netmask) to that format. More flexibility for power users, and ipaddress methods.

Reply:
I will leave it now as it has been for a few years, maybe when you have time you could make this change? :)

Reply:
Could just get rid of that notation altogether when assigning remote_ip - in a way that cannot break actual ipv6 addresses:
```
if remote_ip.startswith("::ffff:") and not remote_ip.find(".") < 0:
    remote_ip = remote_ip.replace("::ffff:", "")
```